corehq/apps/hqadmin/views.py

```diff
@@ -1134,3 +1137,51 @@ def get(self, request, *args, **kwargs):
             return JsonResponse(data)
         else:
             return HttpResponse('Only web users can access this endpoint', status=400)
+
+
+@method_decorator(csrf_exempt, name='dispatch')
+@method_decorator(validate_request_hmac('FORMPLAYER_INTERNAL_AUTH_KEY', ignore_if_debug=True), name='dispatch')
+class SessionDetialsView(View):
+    """
+    Internal API to allow formplayer to get the Django user ID
+    from the session key.
+
+    Authentication is done by HMAC signing of the request body:
+
+        secret = settings.FORMPLAYER_INTERNAL_AUTH_KEY
+        data = '{"session_id": "123"}'
+        digest = base64.b64encode(hmac.new(secret, data, hashlib.sha256).digest())
+        requests.post(url, data=data, headers={'X-MAC-DIGEST': digest})
+    """
+    urlname = 'session_details'
+    http_method_names = ['post']
+
+    def post(self, request, *args, **kwargs):
+        try:
+            data = json.loads(request.body)
+        except ValueError:
+            return HttpResponseBadRequest()
+
+        if not data or not isinstance(data, dict):
+            return HttpResponseBadRequest()
+
+        session_id = data.get('session_id', None)
+        if not session_id:
+            return HttpResponseBadRequest()
+
+        engine = import_module(settings.SESSION_ENGINE)
+        session = engine.SessionStore(session_id)
+        try:
+            if session.is_empty():
+                raise Http404
+        except AttributeError:
+            raise Http404
+
+        user_id = session.get(SESSION_KEY, None)
+        if not user_id:
+            raise Http404
+
+        user_id = get_user_model()._meta.pk.to_python(user_id)
+        return JsonResponse({
+            'user_id': user_id
```

Comment:
it would be nice on the formplayer side if this returned the couch_id as well. it will eliminate us having to do an additional lookup in the user table